src/index.js

```diff
@@ -1,14 +1,31 @@
 import behaviorSubject from 'callbag-behavior-subject'
-import { useRef } from 'react'
+import { useRef, useEffect } from 'react'
 
 export default function useValueCallbag(value) {
-  const value$Ref = useRef(null)
+  const value$Ref = useRef()
+  const lastVal = useRef(value)
 
-  if (value$Ref.current === null) {
+  if (!value$Ref.current) {
     value$Ref.current = behaviorSubject(value)
-  } else {
-    value$Ref.current(1, value)
+    lastVal.current = value
   }
 
-  return value$Ref.current
+  const subject = value$Ref.current
+
+  useEffect(() => {
+    if (lastVal.current !== value) {
+      subject(1, value)
+      lastVal.current = value
+    }
+  }, [value, subject])
+
+  useEffect(
+    () => () => {
+      value$Ref.current(2)
```

Comment:
I'm still not quite sure if we should terminate the source here but the React team doesn't provide any guidance for cases like this 😬 

do we have any realistic use case for which this is needed? I feel like the stuff around the possibility for effects to re-fire based on parent component being hidden/shown influences this quite a bit: https://github.com/facebook/react/issues/13206#issuecomment-805669168 . I'm just unsure yet **how** they influence this. 😅

Reply:
Yeah that is getting a bit shaky 😅 However, I would think (or at least would hope) that we would have a way to run some effects when a component truly is cleaned up.

In terms of use case, I am more concerned about additional _strange_ rules not having this would impose. For example, without this termination, you cannot safely expose these callbags to the outer component (for example, via imperative handles). For example, imagine a generic input component that would expose the stream of its value to outer components:

```jsx
const CustomInput = forwardRef((props, ref) => {
  const [val, set] = useState()
  const stream = useValueCallbag(val)

  useImperativeHandle(ref, () => ({
    stream
  }))

  return <input onInput={e => set(e.target.value)} .../>
})
```
```jsx
function App() {
  const ref = useRef()
  const suggestions = useCallbag([], () => pipe(
    ref.stream,
    debounce(200),
    map(value => fromPromise(fetchSuggestions(value)),
    flatten
  ))

  return <CustomInput ref={ref} suggestions={suggestions} ... />
}
```

Reply:
This doesn't look like something I would want to worry about - React is really all about top->down flow and this breaks it so I would kinda classify this as working against React principles.

IMHO it's better to drop this disposal for now and reevaluate later - when a more compelling use case will arise.

Reply:
makes sense, I removed the disposal code.